src/luarocks/core/cfg.lua

```diff
@@ -372,8 +372,8 @@ local function make_defaults(lua_version, target_cpu, platforms, home)
       defaults.variables.LDFLAGS = os.getenv("LDFLAGS")
 
       defaults.cmake_generator = "Unix Makefiles"
-      defaults.variables.CC = "gcc"
-      defaults.variables.LD = "gcc"
+      defaults.variables.CC = os.getenv("CC") or "gcc"
+      defaults.variables.LD = os.getenv("LD") or "gcc"
```

Comment:
with this patch the build of luasystem fails

LD=ld
```
ld  -shared -o system/core.so src/core.o src/compat.o src/time.o -L/nix/store/6f66prpgx1qx4n6k450sxs3d157ia1ps-glibc-2.35-163/lib -Wl,-rpath,/nix/store/6f66prpgx1qx4n6k450sxs3d157ia1ps-glibc-2.35-163/lib -lrt
/nix/store/vhf2cr6immz4qdxd83y025fnxim8mmfg-binutils-2.38/bin/ld: unrecognized option '-Wl,-rpath,/nix/store/6f66prpgx1qx4n6k450sxs3d157ia1ps-glibc-2.35-163/lib'
/nix/store/vhf2cr6immz4qdxd83y025fnxim8mmfg-binutils-2.38/bin/ld: use the --help option for usage information

Error: Build error: Failed compiling module system/core.so
```

LD=gcc
```
gcc  -shared -o system/core.so src/core.o src/compat.o src/time.o -L/nix/store/...-glibc-2.35-163/lib -Wl,-rpath,/nix/store/...-glibc-2.35-163/lib -lrt
```

`-Wl,rpath... -lrt` are added by https://github.com/luarocks/luarocks/blob/ecb63347bf7d44c68210befc9d03c013350c7831/src/luarocks/build/builtin.lua#L249-L250

Reply:
The patch is not really the problem. The issue here is that LuaRocks expects a gcc-compatible linker frontend in `variables.LD`. You can try something like: LD=`gcc -fuse-ld=gold`. The linker you choose still needs to support the `-rpath` command line option though if  `cfg.gcc_rpath` is set to true.

Reply:
@siffiejoe Indeed, but at this point I think changing this behavior in LuaRocks 3.x would cause more compatibility issues than improvements. We could switch to

```lua
      defaults.variables.LD = os.getenv("CC") or "gcc"
```

and make it a  gradual and less-breaking improvement.

Reply:
> ```lua
>       defaults.variables.LD = os.getenv("CC") or "gcc"
> ```


applied